realm/realm-library/src/androidTest/java/io/realm/RealmIntegerTests.java

```diff
@@ -0,0 +1,326 @@
+     */
+    @Ignore("not yet implemented")
+    @Test
+    public void managed_import() {
+        final RealmInteger ri = new RealmInteger(5);
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.createObject(AllTypes.class).setColumnRealmInteger(ri);
+                    }
+                });
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        RealmInteger ri = realm.where(AllTypes.class).findFirst().getColumnRealmInteger();
+                        ri.set(37);
+                        ri.increment(17);
+                        ri.decrement(7);
+                    }
+                });
+
+        assertEquals(ri.longValue(), 47);
+    }
+
+    /**
+     * Assure that changes to a RealmInteger acquired from a managed object are reflected in the object.
+     */
+    @Ignore("not yet implemented")
+    @Test
+    public void managed_export() {
+        realm.beginTransaction();
+        AllTypes allTypes = realm.createObject(AllTypes.class);
+        allTypes.setColumnRealmInteger(new RealmInteger(5));
+        realm.commitTransaction();
+
+        final RealmInteger ri = allTypes.getColumnRealmInteger();
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        ri.set(37);
+                        ri.increment(17);
+                        ri.decrement(7);
+                    }
+                });
+
+        allTypes = realm.where(AllTypes.class).findFirst();
+        assertEquals(allTypes.getColumnRealmInteger().longValue(), 47);
+    }
+
+    /**
+     * Assure that an unmanaged object's RealmInteger becomes managed when the object does.
+     */
+    @Ignore("not yet implemented")
+    @Test
+    public void managed_copyTo() {
+        final AllTypes allTypes = new AllTypes();
+        RealmInteger ri = new RealmInteger(5);
+        allTypes.setColumnRealmInteger(ri);
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        realm.copyToRealm(allTypes);
+                    }
+                });
+
+        realm.executeTransaction(
+                new Realm.Transaction() {
+                    @Override
+                    public void execute(Realm realm) {
+                        RealmInteger ri = realm.where(AllTypes.class).findFirst().getColumnRealmInteger();
+                        ri.set(37);
+                        ri.increment(17);
+                        ri.decrement(7);
+                    }
+                });
+
+        assertEquals(ri.longValue(), 47);
```

Comment:
This is probably not the semantics we want. Other objects don't magically go between being managed and unmanaged. They have to be copied in either direction.

Thos semantics would be:

```
realm.beginTransaction()
AllTypes obj = realm.createObject(AllTypes.class);
RealmInteger unmaangedRi = new RealmInteger(42);
obj.setRealmInteger(ri)
RaelmInteger managedRi = obj.getRealmInteger();
(managedRi == unmanagedRi) == false;
managedRi.increment(1);
realm.commitTransaction


unmangedRi.get() == 47
unmanagedRi.isValid() == false
unmanagedRi.isManaged() == false
managedRi.get() == 48
managedRi.isValid() == true
managedRi.isManaged() == true
```